scripts/01_prep_inputs/0060-pscis_tidy.R

```diff
@@ -0,0 +1,112 @@
+# and then skip to the re burn to geopackage and csv step
+
+# form_pscis <- sf::st_read(dsn= paste0('../../gis/', dir_project, '/data_field/2023/form_pscis_2023.gpkg')) %>%
+#   # assumes in albers
+#   mutate(
+#     x = sf::st_coordinates(.)[,1],
+#     y = sf::st_coordinates(.)[,2]) %>%
+#   # then grab the utms. fragile since relies on having only 1 utm zone. there
+#   # are functions somewhere to deal with this (can't remember which repo though)
+#   st_transform(crs = 26900 + utm) %>%
+#   mutate(
+#     easting = sf::st_coordinates(.)[,1],
+#     northing = sf::st_coordinates(.)[,2]) %>%
+#   # add in utm zone of study area
+#   mutate(utm_zone = utm) %>%
+#   # not sure we need to but turn non-spatial
+#   sf::st_drop_geometry()
+
+# check for duplicates
+form_pscis %>%
+  filter(!is.na(site_id)) %>%
+  group_by(site_id) %>%
+  filter(n()>1)
+
+# check for sites that have a culvert length over 99.9 or a fill depth over 9.9,
+# anything over this will cause error in submission sheet
+form_pscis %>%
+  filter(length_or_width_meters>99.9|fill_depth_meters>9.9)
+
+form_prep1 <- form_pscis %>%
+  #split date time column into date and time
+  dplyr::mutate(date_time_start = lubridate::ymd_hms(date_time_start),
+                date = lubridate::date(date_time_start),
+                time = hms::as_hms(date_time_start)) %>%
+  # filter out to get only the records newly created
+  filter(!is.na(date_time_start)) %>%
+  mutate(
+    site_id = case_when(is.na(pscis_crossing_id) ~ my_crossing_reference,
+                        T ~ pscis_crossing_id)
+  ) %>%
+  # remove the form making site
+  filter(site_id != '12345') %>%
+  arrange(site_id)
+
+# clean up data fields to make copy and paste to prov template easier
+form_prep2 <- form_prep1 %>%
+  # some columns that have yes/no answers have NA values in mergin, need to change to No
+  # need to add 'No' as default values to mergin
+  mutate(across(contains('yes_no'), ~replace_na(.,'No'))) %>%
+  # some numeric fields for CBS have NA values when a user input 0
+  mutate(across(c(outlet_drop_meters, outlet_pool_depth_0_01m, culvert_slope_percent, stream_slope),
+                ~case_when(crossing_type == 'Closed Bottom Structure' ~replace_na(.,0),
+                           TRUE ~ .
+                ))) %>%
+  # change 'trib' to long version 'Tributary'
+  mutate(stream_name = str_replace_all(stream_name, 'Trib ', 'Tributary ')) %>%
+  # change 'Hwy' to 'Highway'
+  mutate(road_name = str_replace_all(road_name, 'Hwy ', 'Highway '))
+
+# add in which phase of assessment the site is in, did the reassessment sites by hand (because there was only 3-4) so thats not in here.
+# This only works when only phase 1 sites have a priority ranking, could use some updating but works for now.
+form_pscis_cleaned <-  form_prep2 %>%
+  mutate(source = case_when(
+    my_priority == 'phase 2' ~ 'phase2',
```

Comment:
we know a site is phase 1 when it has a modelled_crossing_id.  if its phase 2 or reassessment it has a PSCIS Id already

Reply:
okay yes I saw that actually. Do you have a way to decipher phase 2 and reassessments?